consensus/consensus-types/src/block_retrieval.rs

```diff
@@ -46,7 +72,9 @@ impl fmt::Display for BlockRetrievalRequest {
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
 pub enum BlockRetrievalStatus {
     // Successfully fill in the request.
-    Succeeded,
+    SucceededWithEnoughBlocks,
```

Comment:
I don't think we need to differentiate the two, Succeed => 
```
  if target.is_some() {
    last == target
  } else {
    len == num_blocks
  }
  check it forms a chain
```

Reply:
I think it's clearer to distinguish these two cases. `target.is_some()` does not necessarily mean this is a SucceededWithTarget. During block retrieval, we send a number of requests, each of which contains the target. The client does not know how many rounds are timeouts.

Reply:
the server needs to handle the request correspondingly. you need to change the handler https://github.com/diem/diem/blob/main/consensus/src/round_manager.rs#L856 to reflect the API contract

Reply:
also just a note we also can't change it this way, this is not backward compatible